plone/formwidget/namedfile/widget.py

```diff
@@ -64,6 +82,35 @@ class NamedFileWidget(Explicit, file.FileWidget):
     klass = u'named-file-widget'
     value = None  # don't default to a string
 
+    @property
+    def is_upload(self):
+        """True, if value is a uploaded file.
+        """
+        return isinstance(self.value, FileUpload)
+
+    @property
+    def file_upload_id(self):
+        """Store the uploaded file contents with a file_upload_id key in a
+        temporary folder.
+        This is done, so that in case of a form validation error the already
+        uploaded image can be reused.
+        """
+        if self.is_upload:
+            upload_id = uuid.uuid4().hex
+            tmp_folder = getSite().temp_folder  # Acquire from Zope root.
+            if 'uploads' not in tmp_folder:
+                tmp_folder.manage_addFolder('uploads', 'Uploads')
+            uploads = tmp_folder.uploads
+            self.value.seek(0)
+            if not isinstance(getattr(uploads, 'upload_map', False), dict):
+                uploads.upload_map = {}
```

Comment:
Upload map is an attribute on the uploads folder.
No need to use a folder maybe...